tests/config/test_tls.py

```diff
@@ -88,7 +95,7 @@ def test_tls_client_minimum_1_point_3_missing(self):
             self.addCleanup(setattr, SSL, "SSL.OP_NO_TLSv1_3", OP_NO_TLSv1_3)
             assert not hasattr(SSL, "OP_NO_TLSv1_3")
 
-        config = {"federation_client_minimum_tls_version": 1.3}
+        config: JsonDict = {"federation_client_minimum_tls_version": 1.3}
```

Comment:
I'm guessing we need all these `JsonDict`s because 

- read_config wants JsonDict i.e. Dict[str, Any]
- the dict literal has a different deduced type, e.g. Dict[str, float] here
- Dict is invariant so mypy complains

Could maybe change the read_config to accept a Mapping[str, Any]? :shrug:

Reply:
That could work...I think I'd have to change all of them? I'll give it a try.

Reply:
I don't mind either way tbh---don't stress if this turns out to be a PITA

Reply:
I kind of see tackling making `JsonDict` not suck as a future thing we need to do...

Reply:
Yeah, JsonDict sucks. I think my (purely hypothetical) plan there is either explicit validation/parsing with e.g. pydantic or more use of TypedDict